unit_tests/src/test_user_administration.cpp

```diff
@@ -223,4 +229,39 @@ TEST_CASE("user group administration")
             irods::experimental::defer_authentication, env.rodsHost, env.rodsPort, {test_user.name, env.rodsZone}};
         CHECK(clientLoginWithPassword(static_cast<RcComm*>(user_conn), test_user_prop.value.data()) == 0);
     }
+
+    SECTION("#7576: ensure no limitations on proxied groupadmin")
+    {
+        rodsEnv env{};
+        _getRodsEnv(env);
+
+        REQUIRE_NOTHROW(
+            adm::client::modify_user(conn, generic_user, adm::user_type_property{adm::user_type::groupadmin}));
+        REQUIRE(adm::client::type(conn, generic_user) == adm::user_type::groupadmin);
+
+        // Connect to iRODS without authenticating.
+        irods::experimental::client_connection user_conn{env.rodsHost,
+                                                         env.rodsPort,
+                                                         {env.rodsUserName, env.rodsZone}, // (proxy) rodsadmin user
+                                                         {generic_user.name, env.rodsZone}}; // groupadmin user
+
+        // Try to add a group as the proxied groupadmin user.
+        adm::group test_group{"bad_bad_water_group"};
+        REQUIRE_NOTHROW(adm::client::add_group(user_conn, test_group));
+        irods::at_scope_exit group_cleanup{
+            [&conn, &test_group] { CHECK_NOTHROW(adm::client::remove_group(conn, test_group)); }};
+
+        // Create and remove user using groupadmin
+        adm::user groupadmin_created_user{"mr_flips"};
+        REQUIRE_NOTHROW(adm::client::add_user(user_conn, groupadmin_created_user));
+        irods::at_scope_exit created_user_cleanup{[&conn, &groupadmin_created_user] {
+            CHECK_NOTHROW(adm::client::remove_user(conn, groupadmin_created_user));
+        }};
+
+        // Add and remove user from group using groupadmin
+        REQUIRE_NOTHROW(adm::client::add_user_to_group(user_conn, test_group, groupadmin_created_user));
+        irods::at_scope_exit group_add_cleanup{[&user_conn, &test_group, &groupadmin_created_user] {
+            CHECK_NOTHROW(adm::client::remove_user_from_group(user_conn, test_group, groupadmin_created_user));
+        }};
```

Comment:
I think I found a bug? Looking at `test_igroupadmin.py` for reference, it looks like we want group admins to be bound to a group before they can manage it.

Here I can add a user to a group, but cannot remove it, since the groupadmin was never added to the group.

Reply:
Hmm, that does sound like a bug.

Is it only the library that's violating the rules or the library and the icommands that's violating the rules?

Reply:
If you mean a bug in this test, please disregard the rest of this. :)

If you mean a bug with groupadmins... I think that this is not a bug.

groupadmins can add users to a group in exactly 2 situations:
 1. When the groupadmin is a member of the group
 2. When the group has no members

So the very first thing a groupadmin should do after creating a group to manage is add that groupadmin user to the group so that the groupadmin can manage the new group. This is described in the `igroupadmin mkgroup` help text (https://docs.irods.org/4.3.1/icommands/groupadmin/#mkgroup):

> You will need to add yourself to the new group to then
be able to add and remove others (when the group is empty groupadmins are
allowed to add themselves.)

If the groupadmin is not the first member of the new group, the groupadmin cannot manage the group until a user with permission to manage the group adds that groupadmin. So, the test should first add the groupadmin to the group before adding the other user.

Reply:
from the help text - not sure if this clarifies what you're seeing.

the groupadmin can add themselves only to empty groups.

```
mkgroup
 mkgroup groupName[#Zone] (make a new group)
Make a new group.  You will need to add yourself to the new group to then
be able to add and remove others (when the group is empty groupadmins are
allowed to add themselves.)
```

Reply:
jinx.  and i lost.

Reply:
Excellent save! Forgot about that.

Reply:
Is it fine to allow adding a non-groupadmin to a group first?

Reply:
You can do that - but the groupadmin will not be able to add themselves to a non-empty group.

Reply:
rodsadmin can do whatever they want in whatever order.

Reply:
Thinking of creating an issue to make a test and solidify this as a feature?

Reply:
Sure. Put it in an issue and we'll get it handled in 4.3.3.

Reply:
Issue created, see #7707.